components/retro-go/rg_input.c

```diff
@@ -47,12 +47,21 @@ static rg_battery_t battery_state = {0};
 bool rg_input_read_battery_raw(rg_battery_t *out)
 {
     uint32_t raw_value = 0;
+    int raw_value_adc2 = 0;
     bool present = true;
     bool charging = false;
 
 #if RG_BATTERY_DRIVER == 1 /* ADC1 */
     for (int i = 0; i < 4; ++i)
-        raw_value += esp_adc_cal_raw_to_voltage(adc1_get_raw(RG_BATTERY_ADC_CHANNEL), &adc_chars);
```

Comment:
I would suggest adding braces to the loop and the if blocks that contain another if/else block. Nesting such blocks without braces can introduce subtle bugs in my experience, especially if the indenting is lost at one point.